### PR TITLE
Warnlarge

### PR DESCRIPTION
**Pull request by MarvinT**
Really just an extension of #79 but addresses some of #83

### CODE REVIEW

**Comment by MarvinT** [comment]
How's this? Documentation is updated, tests pass, coverage is ok. How's it look?

File: upsetplot/plotting.py [hunk 1]
```diff
@@ -88,6 +88,14 @@ def _aggregate_data(df, subset_size, sum_over):
 def _process_data(df, sort_by, sort_categories_by, subset_size, sum_over):
     df, agg = _aggregate_data(df, subset_size, sum_over)
 
+    if len(agg.index.levels) >= 30:
```
Comment (jnothman) [inline upsetplot/plotting.py]: This may no longer be an issue after #86

**Comment by jnothman** [comment]
There is the capacity to filter on a min and max value, but I've currently got it hard coded to do nothing:
https://github.com/jnothman/UpSetPlot/blob/master/upsetplot/plotting.py#L129-L131

So what we need, rather, is for `_process_data` and `UpSet` to have parameters to specify that `min_value` and perhaps `max_value` (maybe with better names?), and for test cases, and an example, to be added.

**Comment by jnothman** [comment]
Do you want to make those changes?

**Comment by SichongP** [comment]
Sorry I had just seen this PR. In my PR (#90 ) I implemented an argument to set minimum size. It seems that the `agg = agg[np.logical_and(agg >= min_value, agg <= max_value)]` filter comes after `totals` is calculated and does not update it after filters. This will result in incorrect plots as `plot_totals` relies on `totals` to determine bars to plot (ie. filtered subsets will still have bar representations). I can update my PR to use `min_value` and `max_value` but I think filter should move up before `totals` is calculated.

Additionally, I have a piece about filtering data before feeding to UpSet() in my PR for documentation (#91 ). Obviously this could be updated once #90 is approved.

**Comment by jnothman** [comment]
I agree, @SichongP, just removing the data messes up the totals. Closing this PR.

**Pull request closed without merging**